xayn-ai/src/lib.rs

```diff
@@ -38,6 +38,13 @@ pub use crate::{
 #[cfg(test)]
 mod tests;
 
+// we need to export rstest_reuse from the root for it to works.
+// `use rstest_reuse` will trigger clippy::single_component_path_imports
+// that is not possible to silence.
+#[cfg(test)]
+#[allow(unused_imports)]
+pub(crate) use rstest_reuse as rstest_reuse;
```

Comment:
since `rustfmt` wants to remove same name aliases, but we need the alias for the aforementioned reasons, our only option seems to silence the formatting as well.
```suggestion
#[rustfmt::skip]
pub(crate) use rstest_reuse as rstest_reuse;
```